### PR TITLE
Model predict and raw data switches plus logger cleanup

### LINKED ISSUE #37: Predict model from GNSS catalog
I've implemented model prediction software for TART if you go from json files -- as far as I can tell the hdf5 data don't contain this data.
image

I need to polish it a bit before I PR but I hope this will help me calibrate the Rhodes data

### LINKED ISSUE #38: Logging has reverted to ugly debug prints

### LINKED ISSUE #39: Add switch to dump uncalibrated raw data

### PR DESCRIPTION
**Pull request by bennahugo**
Closes #39
Closes #38 
Closes #37 
@tmolteno please see as discussed. I hope I will be able to calibrate Rhodes phases and baselines using this using standard CASA routines. I've double checked that this works on NZ data by overplotting Tigger LSM format catalogs

### CODE REVIEW

File: bin/tart2ms [hunk 1]
```diff
@@ -105,22 +110,28 @@ if __name__ == "__main__":
             "At the moment can only concatenate from JSON or HDF files, not a combination of the two")
 
     if ARGS.json:
-        print("Getting Data from file: {}".format(ARGS.json))
-        print("Writing measurement set '{}'...".format(ARGS.ms))
+        logger.info("Getting Data from file: {}".format(ARGS.json))
+        logger.info("Writing measurement set '{}'...".format(ARGS.ms))
```
Comment (tmolteno) [inline bin/tart2ms:114]: Generic question. Should we be using the logger for output that the user should always see? I'm happy to approve this but we should discuss.

File: tart2ms/tart2ms.py [hunk 2]
```diff
@@ -596,51 +674,64 @@ def __twelveball(direction):
                             f"{np.ceil(obs_length / (rayleigh_crit / sidereal_rate) * 100):.0f}% "
                             f"of the instrument resolution during the course of this observation")
 
+        if fill_model:
+            if not AFRICANUS_DFT_AVAIL:
+                raise RuntimeError("Cannot predict model visibilities. Please install codex-africanus package")
+            if not sources:
+                LOGGER.critical("You have requested to predict a model for GNSS sources, however one or more of "
```
Comment (tmolteno) [inline tart2ms/tart2ms.py:681]: Should use Point sources rather than GNSS sources.

File: tart2ms/tart2ms.py [hunk 3]
```diff
@@ -596,51 +674,64 @@ def __twelveball(direction):
...
+                        "If sources are specified then we expected epochs to be of same size as sources list")
+                model_data = da.zeros_like(dask_data)
+                spwi_chan_freqs = da.from_array(spw_chan_freqs[spw_i])
+                for dataset_i, (epoch_s_i, sources_i) in enumerate(zip(epoch_s, sources)):
+                    # get J2000 RADEC
+                    sources_radec = np.empty((len(sources_i), 2))
+                    for src_i, src in enumerate(sources_i):
+                        name = src['name']
+                        # Convert to J2000
+                        dir_altaz = SkyCoord(alt=src['el']*u.deg, az=src['az']*u.deg, obstime=obstime[0],
```
Comment (tmolteno) [inline tart2ms/tart2ms.py:696]: We should add helper functions for conversion from altaz -> radec, particularly as I would like to move to a JSON formatted sky model being provided on the command line. So perhaps we write something to parse the sky model into a list of  SkyCoord objects.  Perhaps can discuss offline

**Review by tmolteno — COMMENTED** [review]
This is pretty good. I think it would be an improvement to factor out the parsing of the sky-model json into a function. Then allow that JSON to be passed in on the command line explicitly. 

Also need some kind of testbench if possible.

**Comment by bennahugo** [comment]
We could. Question as always is format. I honestly think Json dictionaries
is not the format to use for storing catalog information long term - there
are various other standardized formats for doing this (see e.g. wsclean,
Tigger lsm and bdsf ASCII formats). Json can be intermediate for a web API
but the core format should be stringized inside.

I prefer all stdio to go into a log (and the log should preferably be
written to disk). I will add a filewriter handler to the logger when
running in __main__.

I agree that there should be helper functions for the altaz conversions.
Will add.

**Comment by bennahugo** [comment]
Ok CASA calibration does work at least on the NZ data. The phases improve to some extent and the astrometry is shifted and better fits the provided GNSS positions, so I'm hopeful for the Rhodes data. I should also hopefully be able to disentagle which antennas are not working at the time the data was taken by looking at the noise spread on the amplitudes after correction

**Pull request merged**